"Weird" macOS observations

A few things I've noticed, not sure if bugs or by design, so listing free-form:

1. Launch Amiberry, UI appears in middle of the screen as a window as expected. Load and start a config with "full window", and a new macOS space is launced for the emulator in full screen (which is great!). Press F12 and the UI is now a maximised window on the original space you launced Amiberry from. Is this correct or should the UI appear with dimensions as per initial launch?

2. If you now click the "Restart" option in the UI, we automatically switch to the space with the current emulation frozen, and if you switch manually to the original space the UI has indeed restarted. When you click start to start an emulation, the UI disappears and you have to manually switch to the space which had the frozen emulation previously, but now shows the current running emulation.

Hope that makes sense.

I added the dependencies for mac from the "Compile from source" page and installed cmake.

Executing `cmake --build build -j$(sysctl -n hw.ncpu)` got to 100% then errored with the below.

```
[100%] Building CXX object CMakeFiles/amiberry.dir/external/floppybridge/src/floppybridge_lib.cpp.o
[100%] Building CXX object CMakeFiles/amiberry.dir/src/osdep/macos_authopen.cpp.o
2 warnings generated.
[100%] Linking CXX executable Amiberry.app/Contents/MacOS/Amiberry
ld: warning: ignoring duplicate libraries: '-liconv', '-lpthread'
Undefined symbols for architecture arm64:
  "_libiconv", referenced from:
      utf8_to_latin1_string(std::__1::basic_string_view<char, std::__1::char_traits<char>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>&) in amiberry_filesys.cpp.o
      utf8_to_latin1_string(std::__1::basic_string_view<char, std::__1::char_traits<char>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>&) in amiberry_filesys.cpp.o
  "_libiconv_close", referenced from:
      utf8_to_latin1_string(std::__1::basic_string_view<char, std::__1::char_traits<char>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>&)::IconvWrapper::~IconvWrapper() in amiberry_filesys.cpp.o
  "_libiconv_open", referenced from:
      void std::__1::__call_once_proxy[abi:ne200100]<std::__1::tuple<utf8_to_latin1_string(std::__1::basic_string_view<char, std::__1::char_traits<char>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>&)::IconvWrapper::get()::'lambda'()&&>>(void*) in amiberry_filesys.cpp.o
ld: symbol(s) not found for architecture arm64
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [Amiberry.app/Contents/MacOS/Amiberry] Error 1
make[1]: *** [CMakeFiles/amiberry.dir/all] Error 2
make: *** [all] Error 2
```

--- a/src/osdep/gui/main_window.cpp
+++ b/src/osdep/gui/main_window.cpp
@@ -1362,9 +1362,33 @@ void amiberry_gui_init()
 			SDL_SyncWindow(mon->gui_window);
 		}
 		if (mon->gui_window) {
-			// Sync rect to actual window size (SDL may adjust)
+			// Keep the intended GUI size — SDL may create the window larger
+			// than requested (e.g. on macOS when returning from a full-window
+			// Space).  We enforce our intended size below instead of blindly
+			// accepting whatever SDL reports.
+			const int intended_w = gui_window_rect.w;
+			const int intended_h = gui_window_rect.h;
 			int ww, wh;
 			SDL_GetWindowSize(mon->gui_window, &ww, &wh);
+
+			SDL_WindowFlags wflags = SDL_GetWindowFlags(mon->gui_window);
+
+			// If the window ended up larger than intended (or maximized/
+			// fullscreen), force it back to the intended dimensions.
+			// This happens on macOS when returning from a full-window Space.
+			if (ww != intended_w || wh != intended_h
+				|| (wflags & (SDL_WINDOW_MAXIMIZED | SDL_WINDOW_FULLSCREEN))) {
+				write_log("GUI window: SDL reports %dx%d (flags=0x%x) but intended %dx%d — correcting\n",
+					ww, wh, (unsigned)wflags, intended_w, intended_h);
+				if (wflags & SDL_WINDOW_MAXIMIZED)
+					SDL_RestoreWindow(mon->gui_window);
+				if (wflags & SDL_WINDOW_FULLSCREEN)
+					SDL_SetWindowFullscreen(mon->gui_window, false);
+				SDL_SetWindowSize(mon->gui_window, intended_w, intended_h);
+				SDL_SyncWindow(mon->gui_window);
+				ww = intended_w;
+				wh = intended_h;
+			}
 			gui_window_rect.w = ww;
 			gui_window_rect.h = wh;
 
@@ -1703,10 +1727,17 @@ void amiberry_gui_halt()
 		// Always save position and size together so the ini stays consistent.
 		// Avoids stale position-only state causing top-left placement on first
 		// launch after upgrading from a version that didn't persist size.
+		// Skip saving dimensions when the window is maximized — the maximized
+		// size is transient (screen-dependent) and would overwrite the user's
+		// preferred normal-window size, causing it to appear maximized on every
+		// subsequent launch.
+		const bool is_maximized = (SDL_GetWindowFlags(mon->gui_window) & SDL_WINDOW_MAXIMIZED) != 0;
 		regsetint(nullptr, _T("GUIPosX"), gui_window_rect.x);
 		regsetint(nullptr, _T("GUIPosY"), gui_window_rect.y);
-		regsetint(nullptr, _T("GUISizeW"), gui_window_rect.w);
-		regsetint(nullptr, _T("GUISizeH"), gui_window_rect.h);
+		if (!is_maximized) {
+			regsetint(nullptr, _T("GUISizeW"), gui_window_rect.w);
+			regsetint(nullptr, _T("GUISizeH"), gui_window_rect.h);
+		}
 #if defined(__ANDROID__)
 		// Don't destroy the window on Android, as we reuse it
 #elif defined(_WIN32)
@@ -1907,9 +1938,12 @@ void run_gui()
 			}
 			else if (gui_event.type == SDL_EVENT_WINDOW_RESIZED
 				&& gui_event.window.windowID == SDL_GetWindowID(mon->gui_window)) {
-				gui_window_rect.w = gui_event.window.data1;
-				gui_window_rect.h = gui_event.window.data2;
-
+				// Don't track size changes caused by the window being maximized —
+				// we only want to persist the user's normal (restored) window size.
+				if (!(SDL_GetWindowFlags(mon->gui_window) & SDL_WINDOW_MAXIMIZED)) {
+					gui_window_rect.w = gui_event.window.data1;
+					gui_window_rect.h = gui_event.window.data2;
+				}
 			}
 			else if (gui_event.type == SDL_EVENT_WINDOW_CLOSE_REQUESTED
 				&& gui_event.window.windowID == SDL_GetWindowID(mon->gui_window)) {

